lib/ejs.js

```diff
@@ -670,147 +695,100 @@ Template.prototype = {
-          switch (this.mode) {
-          case Template.modes.EVAL:
-          case Template.modes.RAW:
-            // HACK: backward-compat `include` preprocessor directives
-            var include;
-            if (this.isAfterTag && (include = line.match(/^\s*include\s+(\S+)/))) {
-              var includeSrc;
-              var includeOpts = utils.shallowCopy({}, self.opts);
-              var includeObj = includeSource(include[1], includeOpts);
-              if (self.opts.compileDebug) {
-                includeSrc =
-                    '    ; (function(){' + '\n'
-                    + '      var __line = 1' + '\n'
-                    + '      , __lines = ' + JSON.stringify(includeObj.template) + '\n'
-                    + '      , __filename = ' + JSON.stringify(includeObj.filename) + ';' + '\n'
-                    + '      try {' + '\n'
-                    + includeObj.source
-                    + '      } catch (e) {' + '\n'
-                    + '        rethrow(e, __lines, __filename, __line, escapeFn);' + '\n'
-                    + '      }' + '\n'
-                    + '    ; }).call(this)' + '\n';
-              }else{
-                includeSrc = '    ; (function(){' + '\n' + includeObj.source +
-                    '    ; }).call(this)' + '\n';
-              }
-              self.source += includeSrc;
-              self.dependencies.push(exports.resolveInclude(include[1],
-                  includeOpts.filename));
-              return; // TODO: debug info
-            }
-            // END HACH include
-            if (line.lastIndexOf('//') > line.lastIndexOf('\n')) {
-              line += '\n';
-            }
-            break;
-          case Template.modes.ESCAPED:
-            if (line.lastIndexOf('//') > line.lastIndexOf('\n')) {
-              line += '\n';
-            }
-          }
-          switch (this.mode) {
-              // Just executing code
-          case Template.modes.EVAL:
-            this.source += '    ; ' + line + '\n';
-            break;
-              // Exec, esc, and output
-          case Template.modes.ESCAPED:
-            this.source += '    ; __append(escapeFn(' + stripSemi(line) + '))' + '\n';
-            break;
-              // Exec and output
-          case Template.modes.RAW:
-            this.source += '    ; __append(' + stripSemi(line) + ')' + '\n';
-            break;
-          case Template.modes.COMMENT:
-                // Do nothing
-            break;
-              // Literal <%% mode, append as raw output
-          case Template.modes.LITERAL:
-            this._addOutput(line);
-            break;
-          }
-        }
-          // In string mode, just add the output
-        else {
-          this._addOutput(line);
-        }
+      break;
+    case Template.modes.ESCAPED:
+      if (line.lastIndexOf('//') > line.lastIndexOf('\n')) {
+        line += '\n';
       }
-      this.isAfterTag = false;
+    }
+
+    switch (this.mode) {
```

Comment:
Unless we're actively using fall-through, might be worth considering doing some sort of dispatch via properties / named methods on an object, e.g.:

```javascript
this.processors = {};
this.processors[Template.modes.EVAL] = function (line) {
  this.source += '    ; ' + line + '\n';
};
// Etc....
var processor = this.processors[this.mode];
if (typeof processor == 'function) {
    processor.call(this, line);
}
```

Something like that. If the function doesn't exist, throw some sort of reasonable error.